File: FormUI/Vote/Mattervote_ins_update.cs
using System;
using System.Data;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace pmDHCD
{
    public partial class Mattervote_ins_update
    {
        private string controlcode;
        private int updatemattercode;
        private int updatedelegatecode;
        private bool isLoading = true;
        public Mattervote_ins_update(string controlcode, int mattercode, int delegatecode)
        {
            this.controlcode = controlcode;
            updatemattercode = mattercode;
            updatedelegatecode = delegatecode;
            // This call is required by the Windows Form Designer.
            InitializeComponent();
            // Add any initialization after the InitializeComponent() call.

        }
        private void Mattervote_ins_update_Load(object sender, EventArgs e)
        {
            isLoading = true;
            MaskedTextBox1.Text = My.MyProject.Forms.Mainform.workingmeeting;
            NumericUpDown1.Value = 1m;

            if (controlcode == "Update")
            {
                Button1.Text = "Cập nhật";
                Text = "Cập nhật phiếu biểu quyết";
                var dt1 = new DataTable();
                try
                {
                    dt1 = My.MyProject.Forms.Mainform.BenlyDal.Matter_getlist(My.MyProject.Forms.Mainform.workingmeeting, updatemattercode);
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi :" + ex.Message);
                    return;
                }
                var dt2 = new DataTable();
                try
                {
                    dt2 = My.MyProject.Forms.Mainform.BenlyDal.Delegate_getlist(My.MyProject.Forms.Mainform.workingmeeting, updatedelegatecode, "");
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi :" + ex.Message);
                    return;
                }
                // NumericUpDown1.Value = CDec(dt1.Rows(0).Item("Mattercode"))
                var val = dt1.Rows[0]["Mattercode"];
                if (!(val is DBNull) && Information.IsNumeric(val.ToString()))
                {
                    NumericUpDown1.Value = Conversions.ToDecimal(val);
                }
                else
                {
                    NumericUpDown1.Value = 1m;
                }
                MaskedTextBox3.Text = Conversions.ToString(dt1.Rows[0]["Mattername"]);
                NumericUpDown1.ReadOnly = true;


                var delegatecode = dt2.Rows[0]["Delegatecode"];
                HolderCodeMaskedTextBox.Text = Conversions.ToString(delegatecode);
                HolderIdentifyMaskedTextBox2.Text = Conversions.ToString(dt2.Rows[0]["IdentityCard"]);
                MaskedTextBox4.Text = Conversions.ToString(dt2.Rows[0]["Delegatename"]);
                HolderCodeMaskedTextBox.ReadOnly = true;
                HolderIdentifyMaskedTextBox2.ReadOnly = true;

                try
                {
                    var dt3 = new DataTable();
                    dt3 = My.MyProject.Forms.Mainform.BenlyDal.MatterVotes_getlist(My.MyProject.Forms.Mainform.workingmeeting, updatemattercode, "", Conversions.ToDecimal(delegatecode));
                    RadioButton1.Checked = Conversions.ToBoolean(dt3.Rows[0]["Agree"]);
                    RadioButton2.Checked = Conversions.ToBoolean(dt3.Rows[0]["DisAgree"]);
                    RadioButton3.Checked = Conversions.ToBoolean(dt3.Rows[0]["Noidea"]);
                    RadioButton3.Checked = Conversions.ToBoolean(dt3.Rows[0]["Illegal"]);
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi :" + ex.Message);
                    return;
                }
                Button2.Visible = false;
                CheckBox1.Visible = false;

            }
            isLoading = false;
        }

        private void NumericUpDown1_ValueChanged(object sender, EventArgs e)
        {
            var dt = new DataTable();

            try
            {
                dt = My.MyProject.Forms.Mainform.BenlyDal.Matter_getlist(My.MyProject.Forms.Mainform.workingmeeting, NumericUpDown1.Value);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi :" + ex.Message);
                return;
            }
            if (dt.Rows.Count == 1)
            {
                MaskedTextBox3.Text = Conversions.ToString(dt.Rows[0]["Mattername"]);
                if (!(dt.Rows[0]["Mattercode"] is DBNull) && Information.IsNumeric(dt.Rows[0]["Mattercode"]))
                {
                    NumericUpDown1.Value = Convert.ToDecimal(dt.Rows[0]["Mattercode"]);
                }
                else
                {
                    NumericUpDown1.Value = 1m;
                }

                if (isLoading)
                    return;
            }
            else
            {
                MaskedTextBox3.Text = "";
            }
        }

        private void MaskedTextBox5_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                var dt = new DataTable();
                try
                {
                    dt = My.MyProject.Forms.Mainform.BenlyDal.Delegate_getlist(My.MyProject.Forms.Mainform.workingmeeting, Conversions.ToDecimal(HolderCodeMaskedTextBox.Text), "");
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi :" + ex.Message);
                    return;
                }
                if (dt.Rows.Count == 1)
                {
                    HolderCodeMaskedTextBox.Text = Conversions.ToString(dt.Rows[0]["Delegatecode"]);
                    HolderIdentifyMaskedTextBox2.Text = Conversions.ToString(dt.Rows[0]["IdentityCard"]);
                    MaskedTextBox4.Text = Conversions.ToString(dt.Rows[0]["Delegatename"]);
                    StockTextBox1.Text = Conversions.ToString(dt.Rows[0]["voterights"]);
                }
                else
                {
                    HolderCodeMaskedTextBox.Text = "";
                    HolderIdentifyMaskedTextBox2.Text = "";
                    MaskedTextBox4.Text = "";
                    StockTextBox1.Text = "";
                }

                if (CheckBox1.Checked == true)
                {
                    insert();
                    HolderCodeMaskedTextBox.Focus();
                    HolderCodeMaskedTextBox.SelectAll();
                }
                else
                {
                    Button1.Focus();
                }

            }
        }


        private void MaskedTextBox5_Leave(object sender, EventArgs e)
        {
            var dt = new DataTable();
            if (string.IsNullOrEmpty(HolderCodeMaskedTextBox.Text))
            {
                return;
            }
            try
            {
                dt = My.MyProject.Forms.Mainform.BenlyDal.Delegate_getlist(My.MyProject.Forms.Mainform.workingmeeting, Conversions.ToDecimal(HolderCodeMaskedTextBox.Text), "");
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi :" + ex.Message);
                return;
            }
            if (dt.Rows.Count == 1)
            {
                HolderCodeMaskedTextBox.Text = Conversions.ToString(dt.Rows[0]["Delegatecode"]);
                HolderIdentifyMaskedTextBox2.Text = Conversions.ToString(dt.Rows[0]["IdentityCard"]);
                MaskedTextBox4.Text = Conversions.ToString(dt.Rows[0]["Delegatename"]);
                StockTextBox1.Text = Conversions.ToString(dt.Rows[0]["voterights"]);
            }
            else
            {
                HolderCodeMaskedTextBox.Text = "";
                HolderIdentifyMaskedTextBox2.Text = "";
                MaskedTextBox4.Text = "";
                StockTextBox1.Text = "";
            }

        }

        private void insert()
        {
            try
            {
                string meetingcode = My.MyProject.Forms.Mainform.workingmeeting;
                decimal mattercode = NumericUpDown1.Value;
                decimal HolderCode = 0m; // CDec(HolderCodeMaskedTextBox.Text)
                decimal DelegateCode = Conversions.ToDecimal(HolderCodeMaskedTextBox.Text);
                bool Agree;
                bool disAgree;
                bool noidea;
                bool illegal;
                My.MyProject.Forms.Mainform.BenlyDal.MatterVotes_insert(meetingcode, mattercode, HolderCode, DelegateCode, RadioButton1.Checked, RadioButton2.Checked, RadioButton3.Checked, rbillegal.Enabled);
            }

            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi :" + ex.Message);
            }
        }
        private void Updatemattervote()
        {
            try
            {
                My.MyProject.Forms.Mainform.BenlyDal.MatterVotes_update(My.MyProject.Forms.Mainform.workingmeeting, NumericUpDown1.Value, Conversions.ToDecimal(HolderCodeMaskedTextBox.Text), RadioButton1.Checked, RadioButton2.Checked, RadioButton3.Checked, rbillegal.Checked);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi :" + ex.Message);
            }
        }
        private void MaskedTextBox2_KeyDown(object sender, KeyEventArgs e)
        {

            if (e.KeyCode == Keys.Enter)
            {

                if (HolderIdentifyMaskedTextBox2.Text is not null && !string.IsNullOrEmpty(HolderIdentifyMaskedTextBox2.Text))
                {
                    TimDaiBieu();
                }

                // Dim dt As New DataTable
                // Try
                // dt = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, 0, MaskedTextBox2.Text)
                // Catch ex As Exception
                // MsgBox("Lỗi :" + ex.Message)
                // Exit Sub
                // End Try
                // If dt.Rows.Count = 1 Then
                // MaskedTextBox5.Text = dt.Rows(0).Item("Delegatecode")
                // MaskedTextBox2.Text = dt.Rows(0).Item("IdentityCard")
                // MaskedTextBox4.Text = dt.Rows(0).Item("Delegatename")
                // StockTextBox1.Text = dt.Rows(0).Item("voterights")
                // Else
                // MaskedTextBox5.Text = ""
                // MaskedTextBox2.Text = ""
                // MaskedTextBox4.Text = ""
                // StockTextBox1.Text = ""
                // End If

                if (CheckBox1.Checked == true)
                {
                    insert();
                    HolderCodeMaskedTextBox.Focus();
                }
                else
                {
                    Button1.Focus();
                }
            }
        }

        private void TimDaiBieu()
        {
            var daibieu = new DataTable();
            try
            {
                // daibieu = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, 0, MaskedTextBox3.Text)
                // daibieu = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, 0, MaskedTextBox2.Text)
                daibieu = My.MyProject.Forms.Mainform.BenlyDal.Authorizations_getlist(My.MyProject.Forms.Mainform.workingmeeting, 0m, "", delegateIdentityTextBox.Text, HolderIdentifyMaskedTextBox2.Text);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi" + ex.Message);
                return;
            }
            if (daibieu.Rows.Count == 1)
            {
                HolderCodeMaskedTextBox.Text = Conversions.ToString(daibieu.Rows[0]["HolderCode"]);
                delegateCodeMaskedTextBox.Text = Conversions.ToString(daibieu.Rows[0]["DelegateCode"]);
                HolderIdentifyMaskedTextBox2.Text = Conversions.ToString(daibieu.Rows[0]["HolderIdentity"]);
                MaskedTextBox4.Text = Conversions.ToString(daibieu.Rows[0]["HolderName"]);
                StockTextBox1.Text = Conversions.ToString(daibieu.Rows[0]["Delegateright"]);
                delegateIdentityTextBox.Text = ""; // daibieu.Rows(0).Item("IdentityCard")
                delegateNameTextbox.Text = Conversions.ToString(daibieu.Rows[0]["DelegateName"]);
            }

            else if (daibieu.Rows.Count > 1)
            {
                var objListForSelect = new AuthorizationListForSelect();
                objListForSelect.HolderIdentifyCard = HolderIdentifyMaskedTextBox2.Text;
                objListForSelect.DelegateIdentifyCard = delegateIdentityTextBox.Text;
                objListForSelect.ShowDialog();
                HolderIdentifyMaskedTextBox2.Text = objListForSelect.DataGridView1.CurrentRow.Cells["HolderIdentity"].Value.ToString();
                delegateIdentityTextBox.Text = objListForSelect.DataGridView1.CurrentRow.Cells["IdentityCard"].Value.ToString();
                TimDaiBieu();
            }
            else if (daibieu.Rows.Count == 0)
            {
                MessageBox.Show("Không tìm thấy đại biểu !");
                HolderCodeMaskedTextBox.Text = "";
                HolderIdentifyMaskedTextBox2.Text = "";
                MaskedTextBox4.Text = "";
                StockTextBox1.Text = "";
                HolderCodeMaskedTextBox.Focus();
                HolderCodeMaskedTextBox.SelectAll();
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(HolderCodeMaskedTextBox.Text))
            {
                HolderCodeMaskedTextBox.Focus();
                return;
            }
            if (controlcode == "Add")
            {

                insert();
                Button2.Focus();
                Button2.Enabled = true;
                Button1.Enabled = false;
            }
            else if (controlcode == "Update")
            {
                Updatemattervote();
                Close();
            }

        }

        private void Button2_Click(object sender, EventArgs e)
        {
            HolderCodeMaskedTextBox.Text = "";
            HolderIdentifyMaskedTextBox2.Text = "";
            MaskedTextBox4.Text = "";
            StockTextBox1.Text = "";
            delegateNameTextbox.Text = "";
            HolderCodeMaskedTextBox.Focus();
            Button1.Enabled = true;
            Button2.Enabled = false;
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            var f = new MatterVotes_ins_remain((int)Math.Round(NumericUpDown1.Value));
            f.ShowDialog();
        }
        private void Button4_Click(object sender, EventArgs e)
        {
            Close();
        }
        private void MaskedTextBox5_MaskInputRejected(object sender, MaskInputRejectedEventArgs e)
        {

        }

        private void Mattervote_ins_update_KeyUp(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
            {
                Close();
            }
        }

        private void NumericUpDown1_Validating(object sender, System.ComponentModel.CancelEventArgs e)
        {
            if (string.IsNullOrEmpty(NumericUpDown1.Text))
            {
                NumericUpDown1.Value = 1m;
            }
        }
    }
}

File: FormUI/Vote/Mattervote_ins_update.vb
Public Class Mattervote_ins_update
    Dim controlcode As String
    Dim updatemattercode As Integer
    Dim updatedelegatecode As Integer
    Dim isLoading As Boolean = True
    Public Sub New(ByVal controlcode As String, ByVal mattercode As Integer, ByVal delegatecode As Integer)
        Me.controlcode = controlcode
        Me.updatemattercode = mattercode
        Me.updatedelegatecode = delegatecode
        ' This call is required by the Windows Form Designer.
        InitializeComponent()
        ' Add any initialization after the InitializeComponent() call.

    End Sub
    Private Sub Mattervote_ins_update_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        isLoading = True
        MaskedTextBox1.Text = Mainform.workingmeeting
        NumericUpDown1.Value = CDec(1)

        If controlcode = "Update" Then
            Button1.Text = "Cập nhật"
            Me.Text = "Cập nhật phiếu biểu quyết"
            Dim dt1 As New DataTable
            Try
                dt1 = Mainform.BenlyDal.Matter_getlist(Mainform.workingmeeting, updatemattercode)
            Catch ex As Exception
                MsgBox("Lỗi :" + ex.Message)
                Exit Sub
            End Try
            Dim dt2 As New DataTable
            Try
                dt2 = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, updatedelegatecode, "")
            Catch ex As Exception
                MsgBox("Lỗi :" + ex.Message)
                Exit Sub
            End Try
            'NumericUpDown1.Value = CDec(dt1.Rows(0).Item("Mattercode"))
            Dim val = dt1.Rows(0).Item("Mattercode")
            If Not IsDBNull(val) AndAlso IsNumeric(val.ToString()) Then
                NumericUpDown1.Value = CDec(val)
            Else
                NumericUpDown1.Value = 1
            End If
            MaskedTextBox3.Text = dt1.Rows(0).Item("Mattername")
            NumericUpDown1.ReadOnly = True


            Dim delegatecode = dt2.Rows(0).Item("Delegatecode")
            HolderCodeMaskedTextBox.Text = delegatecode
            HolderIdentifyMaskedTextBox2.Text = dt2.Rows(0).Item("IdentityCard")
            MaskedTextBox4.Text = dt2.Rows(0).Item("Delegatename")
            HolderCodeMaskedTextBox.ReadOnly = True
            HolderIdentifyMaskedTextBox2.ReadOnly = True

            Try
                Dim dt3 As New DataTable
                dt3 = Mainform.BenlyDal.MatterVotes_getlist(Mainform.workingmeeting, updatemattercode, "", delegatecode)
                RadioButton1.Checked = dt3.Rows(0).Item("Agree")
                RadioButton2.Checked = dt3.Rows(0).Item("DisAgree")
                RadioButton3.Checked = dt3.Rows(0).Item("Noidea")
                RadioButton3.Checked = dt3.Rows(0).Item("Illegal")
            Catch ex As Exception
                MsgBox("Lỗi :" + ex.Message)
                Exit Sub
            End Try
            Button2.Visible = False
            CheckBox1.Visible = False

        End If
        isLoading = False
    End Sub

    Private Sub NumericUpDown1_ValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles NumericUpDown1.ValueChanged
        Dim dt As New DataTable

        Try
            dt = Mainform.BenlyDal.Matter_getlist(Mainform.workingmeeting, NumericUpDown1.Value)
        Catch ex As Exception
            MsgBox("Lỗi :" + ex.Message)
            Exit Sub
        End Try
        If dt.Rows.Count = 1 Then
            MaskedTextBox3.Text = dt.Rows(0).Item("Mattername")
            If Not IsDBNull(dt.Rows(0).Item("Mattercode")) AndAlso IsNumeric(dt.Rows(0).Item("Mattercode")) Then
                NumericUpDown1.Value = Convert.ToDecimal(dt.Rows(0).Item("Mattercode"))
            Else
                NumericUpDown1.Value = 1
            End If

            If isLoading Then Exit Sub
        Else
            MaskedTextBox3.Text = ""
        End If
    End Sub

    Private Sub MaskedTextBox5_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles HolderCodeMaskedTextBox.KeyDown
        If e.KeyCode = Keys.Enter Then
            Dim dt As New DataTable
            Try
                dt = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, HolderCodeMaskedTextBox.Text, "")
            Catch ex As Exception
                MsgBox("Lỗi :" + ex.Message)
                Exit Sub
            End Try
            If dt.Rows.Count = 1 Then
                HolderCodeMaskedTextBox.Text = dt.Rows(0).Item("Delegatecode")
                HolderIdentifyMaskedTextBox2.Text = dt.Rows(0).Item("IdentityCard")
                MaskedTextBox4.Text = dt.Rows(0).Item("Delegatename")
                StockTextBox1.Text = dt.Rows(0).Item("voterights")
            Else
                HolderCodeMaskedTextBox.Text = ""
                HolderIdentifyMaskedTextBox2.Text = ""
                MaskedTextBox4.Text = ""
                StockTextBox1.Text = ""
            End If

            If CheckBox1.Checked = True Then
                insert()
                HolderCodeMaskedTextBox.Focus()
                HolderCodeMaskedTextBox.SelectAll()
            Else
                Button1.Focus()
            End If

        End If
    End Sub


    Private Sub MaskedTextBox5_Leave(ByVal sender As Object, ByVal e As System.EventArgs) Handles HolderCodeMaskedTextBox.Leave
        Dim dt As New DataTable
        If String.IsNullOrEmpty(HolderCodeMaskedTextBox.Text) Then
            Exit Sub
        End If
        Try
            dt = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, HolderCodeMaskedTextBox.Text, "")
        Catch ex As Exception
            MsgBox("Lỗi :" + ex.Message)
            Exit Sub
        End Try
        If dt.Rows.Count = 1 Then
            HolderCodeMaskedTextBox.Text = dt.Rows(0).Item("Delegatecode")
            HolderIdentifyMaskedTextBox2.Text = dt.Rows(0).Item("IdentityCard")
            MaskedTextBox4.Text = dt.Rows(0).Item("Delegatename")
            StockTextBox1.Text = dt.Rows(0).Item("voterights")
        Else
            HolderCodeMaskedTextBox.Text = ""
            HolderIdentifyMaskedTextBox2.Text = ""
            MaskedTextBox4.Text = ""
            StockTextBox1.Text = ""
        End If

    End Sub

    Private Sub insert()
        Try
            Dim meetingcode As String = Mainform.workingmeeting
            Dim mattercode As Decimal = NumericUpDown1.Value
            Dim HolderCode As Decimal = 0 'CDec(HolderCodeMaskedTextBox.Text)
            Dim DelegateCode As Decimal = CDec(HolderCodeMaskedTextBox.Text)
            Dim Agree As Boolean
            Dim disAgree As Boolean
            Dim noidea As Boolean
            Dim illegal As Boolean
            Mainform.BenlyDal.MatterVotes_insert(meetingcode, mattercode, HolderCode, DelegateCode, RadioButton1.Checked, RadioButton2.Checked, RadioButton3.Checked, rbillegal.Enabled)

        Catch ex As Exception
            MsgBox("Lỗi :" + ex.Message)
        End Try
    End Sub
    Private Sub Updatemattervote()
        Try
            Mainform.BenlyDal.MatterVotes_update(Mainform.workingmeeting, NumericUpDown1.Value, HolderCodeMaskedTextBox.Text, RadioButton1.Checked, RadioButton2.Checked, RadioButton3.Checked, rbillegal.Checked)
        Catch ex As Exception
            MsgBox("Lỗi :" + ex.Message)
        End Try
    End Sub
    Private Sub MaskedTextBox2_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles HolderIdentifyMaskedTextBox2.KeyDown

        If e.KeyCode = Keys.Enter Then

            If Not HolderIdentifyMaskedTextBox2.Text Is Nothing AndAlso HolderIdentifyMaskedTextBox2.Text <> "" Then
                Me.TimDaiBieu()
            End If

            'Dim dt As New DataTable
            'Try
            '    dt = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, 0, MaskedTextBox2.Text)
            'Catch ex As Exception
            '    MsgBox("Lỗi :" + ex.Message)
            '    Exit Sub
            'End Try
            'If dt.Rows.Count = 1 Then
            '    MaskedTextBox5.Text = dt.Rows(0).Item("Delegatecode")
            '    MaskedTextBox2.Text = dt.Rows(0).Item("IdentityCard")
            '    MaskedTextBox4.Text = dt.Rows(0).Item("Delegatename")
            '    StockTextBox1.Text = dt.Rows(0).Item("voterights")
            'Else
            '    MaskedTextBox5.Text = ""
            '    MaskedTextBox2.Text = ""
            '    MaskedTextBox4.Text = ""
            '    StockTextBox1.Text = ""
            'End If

            If CheckBox1.Checked = True Then
                insert()
                HolderCodeMaskedTextBox.Focus()
            Else
                Button1.Focus()
            End If
        End If
    End Sub

    Private Sub TimDaiBieu()
        Dim daibieu As New DataTable
        Try
            'daibieu = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, 0, MaskedTextBox3.Text)
            'daibieu = Mainform.BenlyDal.Delegate_getlist(Mainform.workingmeeting, 0, MaskedTextBox2.Text)
            daibieu = Mainform.BenlyDal.Authorizations_getlist(Mainform.workingmeeting, 0, "", delegateIdentityTextBox.Text, HolderIdentifyMaskedTextBox2.Text)
        Catch ex As Exception
            MsgBox("Lỗi" + ex.Message)
            Exit Sub
        End Try
        If daibieu.Rows.Count = 1 Then
            HolderCodeMaskedTextBox.Text = daibieu.Rows(0).Item("HolderCode")
            delegateCodeMaskedTextBox.Text = daibieu.Rows(0).Item("DelegateCode")
            HolderIdentifyMaskedTextBox2.Text = daibieu.Rows(0).Item("HolderIdentity")
            MaskedTextBox4.Text = daibieu.Rows(0).Item("HolderName")
            StockTextBox1.Text = daibieu.Rows(0).Item("Delegateright")
            delegateIdentityTextBox.Text = "" 'daibieu.Rows(0).Item("IdentityCard")
            delegateNameTextbox.Text = daibieu.Rows(0).Item("DelegateName")

        ElseIf daibieu.Rows.Count > 1 Then
            Dim objListForSelect As New AuthorizationListForSelect()
            objListForSelect.HolderIdentifyCard = HolderIdentifyMaskedTextBox2.Text
            objListForSelect.DelegateIdentifyCard = delegateIdentityTextBox.Text
            objListForSelect.ShowDialog()
            HolderIdentifyMaskedTextBox2.Text = objListForSelect.DataGridView1.CurrentRow.Cells("HolderIdentity").Value.ToString()
            delegateIdentityTextBox.Text = objListForSelect.DataGridView1.CurrentRow.Cells("IdentityCard").Value.ToString()
            TimDaiBieu()
        ElseIf daibieu.Rows.Count = 0 Then
            MessageBox.Show("Không tìm thấy đại biểu !")
            HolderCodeMaskedTextBox.Text = ""
            HolderIdentifyMaskedTextBox2.Text = ""
            MaskedTextBox4.Text = ""
            StockTextBox1.Text = ""
            HolderCodeMaskedTextBox.Focus()
            HolderCodeMaskedTextBox.SelectAll()
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If String.IsNullOrEmpty(HolderCodeMaskedTextBox.Text) Then
            HolderCodeMaskedTextBox.Focus()
            Exit Sub
        End If
        If controlcode = "Add" Then

            insert()
            Button2.Focus()
            Button2.Enabled = True
            Button1.Enabled = False
        ElseIf controlcode = "Update" Then
            Updatemattervote()
            Me.Close()
        End If

    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        HolderCodeMaskedTextBox.Text = ""
        HolderIdentifyMaskedTextBox2.Text = ""
        MaskedTextBox4.Text = ""
        StockTextBox1.Text = ""
        delegateNameTextbox.Text = ""
        HolderCodeMaskedTextBox.Focus()
        Button1.Enabled = True
        Button2.Enabled = False
    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        Dim f As New MatterVotes_ins_remain(NumericUpDown1.Value)
        f.ShowDialog()
    End Sub
    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button4.Click
        Me.Close()
    End Sub
    Private Sub MaskedTextBox5_MaskInputRejected(ByVal sender As System.Object, ByVal e As System.Windows.Forms.MaskInputRejectedEventArgs)

    End Sub

    Private Sub Mattervote_ins_update_KeyUp(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles MyBase.KeyUp
        If e.KeyCode = Keys.Escape Then
            Me.Close()
        End If
    End Sub

    Private Sub NumericUpDown1_Validating(sender As Object, e As System.ComponentModel.CancelEventArgs) Handles NumericUpDown1.Validating
        If NumericUpDown1.Text = "" Then
            NumericUpDown1.Value = CDec(1)
        End If
    End Sub
End Class